pkg/edgeview/src/network.go

```diff
@@ -0,0 +1,1243 @@
+				printColor(str2, colorRED)
+			} else {
+				fmt.Printf("%s", str2)
+			}
+		}
+		i++
+	}
+}
+
+// getARP
+func getARP(opt string) {
+	outbytes, err := ioutil.ReadFile("/proc/net/arp")
+	if err != nil {
+		return
+	}
+
+	printColor(" - arp entries: \n", colorBLUE)
+
+	lines := bytes.Split(outbytes, []byte("\n"))
+	var intfarps []intfIP
+	for _, l := range lines {
+		ele := bytes.Fields(l)
+		if len(ele) != 6 {
+			continue
+		}
+		// IP address       HW type     Flags       HW address            Mask     Device
+		// 10.1.0.2         0x1         0x2         00:16:3e:00:01:02     *        bn1
+		var complete bool
+		if string(ele[2]) == "0x2" {
+			complete = true
+		}
+		item := intfIP{
+			intfName: string(ele[5]),
+			ipAddr:   string(ele[0]),
+			hwAddr:   string(ele[3]),
+			arpComplete: complete,
+		}
+		intfarps = append(intfarps, item)
+	}
+
+	for _, i := range intfarps {
+		if (opt != "" &&
+			(strings.Contains(i.intfName, opt) || strings.Contains(i.ipAddr, opt) || strings.Contains(i.hwAddr, opt))) {
+				continue
+		}
+		fmt.Printf("%s, %s, %s, complete %v\n", i.intfName, i.ipAddr, i.hwAddr, i.arpComplete)
+	}
+}
+
+// runACLs
+func runACLs(isRunningACL bool, filter string) {
+	acltables := []string{"raw", "filter", "nat", "mangle"}
+	for _, tbl := range acltables {
+		if filter != "" && filter != tbl {
+			continue
+		}
+		var op string
+		if isRunningACL {
+			printColor(" Configured iptables: "+tbl, colorCYAN)
+			op = "-S"
+		} else {
+			printColor(" Installed iptables: "+tbl, colorCYAN)
+			op = "-nvL"
+		}
+		_, _ = runCmd("iptables "+op+" -t "+tbl, true)
+	}
+}
+
+// runRoute
+func runRoute() {
+	rules, err := netlink.RuleList(syscall.AF_INET)
+	if err != nil {
+		fmt.Printf("runRoute: rule error %v", err)
+		return
+	}
+
+	printColor(" - ip rule:", colorRED)
+	for _, rule := range rules {
+		fmt.Printf("%d: table %d, %s\n", rule.Priority, rule.Table, rule.String())
+	}
+
+	upIntfs := allUPIntfIPv4()
+	for _, i := range upIntfs {
+		link, err := netlink.LinkByName(i.intfName)
+		if err != nil {
+			continue
+		}
+		routes := getAllIPv4Routes(link.Attrs().Index)
+
+		printColor("show route in interfaces: "+i.intfName, colorCYAN)
+		for _, r := range routes {
+			fmt.Printf("%s\n", r.String())
+		}
+	}
+
+	printColor(" - ip fib trie:", colorRED)
+	retbytes, err := ioutil.ReadFile("/proc/net/fib_trie")
+	if err == nil {
+		fmt.Printf("%s\n", string(retbytes))
+	}
+}
+
+func getTables(rules string) []string {
+	rulelines := strings.Split(rules, "\n")
+	var t []string
+	n := len(rulelines)
+	for _, rule := range rulelines[:n-1] {
+		table := strings.Split(rule, "lookup ")
+		if len(table) > 0 {
+			if !strings.Contains(table[1], "default") {
+				t = append(t, table[1])
+			}
+		}
+	}
+	return t
+}
+
+func getMetricsMap(path string, stats *urlStats, isPrint bool) {
+	retbytes, err := ioutil.ReadFile(path+"global.json")
+	if err != nil {
+		return
+	}
+	pathname := ""
+	paths := strings.Split(path, "/")
+	if len(paths) > 3 {
```

Comment:
What output does this produce when the len is shorter than 3?

Reply:
changed to len(paths) != 3, then bail out.